src/Crypto.gren

```diff
@@ -0,0 +1,2227 @@
+
+The returned `Task` should not fail. If it does, please file a ticket!
+-}
+signWithHmac : Key HmacKey HmacKeyParams -> Bytes -> Task x Signature
+signWithHmac (Key key _) bytes =
+    Gren.Kernel.Crypto.signWithHmac 
+        key 
+        bytes
+
+
+
+-- VERIFY
+
+
+{-| Verify that some `Bytes` were signed with the passed `Signature` with the 
+RSA-SSAPKCS1v1.5 algorithm.
+
+This function produces no values. Instead, the `Task` succeeds if the passed 
+signature is valid and fails otherwise.
+-}
+verifyWithRsaSsaPkcs1V1_5 : PublicKey RsaSsaPkcs1V1_5Key RsaKeyParams -> Signature -> Bytes -> Task x a
+verifyWithRsaSsaPkcs1V1_5 (PublicKey (Key key _)) signature bytes =
+    Gren.Kernel.Crypto.verifyWithRsaSsaPkcs1V1_5
+        key
+        signature
+        bytes
+
+
+{-| Verify that some `Bytes` were signed with the passed `Signature` with the 
+RSA-PSS algorithm.
+
+This function produces no values. Instead, the `Task` succeeds if the passed 
+signature is valid and fails otherwise.
+-}
+verifyWithRsaPss : RsaPssParams -> PublicKey RsaPssKey RsaKeyParams -> Signature -> Bytes -> Task x a
+verifyWithRsaPss { salt } (PublicKey (Key key _)) signature bytes =
+    Gren.Kernel.Crypto.verifyWithRsaPss
+        salt
+        key
+        signature
+        bytes
+
+
+{-| Verify that some `Bytes` were signed with the passed `Signature` with the 
+ECDSA algorithm.
+
+This function produces no values. Instead, the `Task` succeeds if the passed 
+signature is valid and fails otherwise.
+-}
+verifyWithEcdsa : DigestAlgorithm -> PublicKey EcdsaKey EcKeyParams -> Signature -> Bytes -> Task x a
+verifyWithEcdsa hash (PublicKey (Key key _)) signature bytes =
+    Gren.Kernel.Crypto.verifyWithEcdsa
+        (digestAlgorithmToString hash)
+        key
+        signature
+        bytes
+
+
+{-| Verify that some `Bytes` were signed with the passed `Signature` with the 
+HMAC algorithm.
+
+This function produces no values. Instead, the `Task` succeeds if the passed 
+signature is valid and fails otherwise.
+-}
+verifyWithHmac : Key HmacKey HmacKeyParams -> Signature -> Bytes -> Task x a
+verifyWithHmac (Key key _) signature bytes =
+    Gren.Kernel.Crypto.verifyWithHmac
+        key
+        signature
+        bytes
+
+
+
+-- DIGEST
+
+
+{-| Supported algorithms suitable for digesting data.
+
+Note: The algorithm `SHA1` is supported by the WebCrypto API, but not available 
+in this package due to known security vunerabilities.
+-}
+type DigestAlgorithm
+    = Sha256
+    | Sha384
+    | Sha512
+
+
+{-| Take some `Bytes` and create a hash from them using the passed `DigestAlgorithm`.
+This operation should always succeed.
+-}
+digest : DigestAlgorithm -> Bytes -> SecureContext -> Task x Bytes
+digest algorithm data _context =
+    Gren.Kernel.Crypto.digest
+        (digestAlgorithmToString algorithm)
+        data
+
+
+
+-- UTILITIES
+
+
+{-|-}
+clamp : Int -> Int -> Int -> Int
+clamp min max number =
```

Comment:
This function already exists in `Basics`: https://packages.gren-lang.org/package/gren-lang/core/version/5.0.2/module/Basics#clamp